### PR TITLE
Add Sprite & SpriteList programming guide pages

### PR DESCRIPTION
**Pull request by pushfoo**
Stacks onto #1724; second PR split from #1620.

### Changes

Adds a structure similar to the one outlined in #1194, consisting of:

* A landing page for sprites & drawing topics in the programming guide
* An overview of Sprite & SpriteList concepts
* A bare minimum drawing example
* An advanced techniques page
* Links to the new pages from the programming guide & front page indices

### How to test

1. Clone the branch
2. Run `python make.py serve`
3. Navigate to the new pages in your browser

### CODE REVIEW

File: doc/example_code/how_to_examples/index.rst [hunk 1]
```diff
@@ -403,6 +404,8 @@ Backgrounds
 
    :ref:`background_parallax`
 
+.. _examples_cameras:
+
```
Comment (pvcraven) [inline doc/example_code/how_to_examples/index.rst:408]: Should we be consistent in naming? 'sprite_examples' and 'camera_examples'?

File: doc/programming_guide/sprites/index.rst [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+   :maxdepth: 1
+
+   spritelists
+   advanced
+
+
+I'm Impatient!
+--------------
+
+Beginners should at least skim the , but you
```
Comment (pvcraven) [inline doc/programming_guide/sprites/index.rst]: Skim the what?

File: doc/programming_guide/sprites/spritelists.rst [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            self.circle.position = self.width // 2, self.height // 2  # Put it in the middle
+            self.sprites.append(self.circle)  # Append the instance to the SpriteList
+
+        def on_draw(self):
+            # 3. Call draw() on the SpriteList inside an on_draw() method
+            self.sprites.draw()
+
+
+    game = WhiteSpriteCircleExample()
+    game.run()
```
Comment (pvcraven) [inline doc/programming_guide/sprites/spritelists.rst]: Might think of pulling this out as a separate file? Then the file could be run as part of the examples test?

File: doc/programming_guide/sprites/spritelists.rst [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+Beginners should see the following to learn more, such as how to load images
+into sprites:
+
+* :ref:`Arcade's Sprite examples <sprite_examples>`
+* :ref:`Arcade's Simple Platformer Tutorial <platformer_tutorial>`
+* The :py:class:`~arcade.Sprite` API documentation
+
+
+Viewports, Cameras, and Screens!
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (pvcraven) [inline doc/programming_guide/sprites/spritelists.rst]: I'd keep the punctuation out of the title

**Review by pvcraven — CHANGES_REQUESTED** [review]
Looking good!

File: doc/example_code/how_to_examples/index.rst (hunk 1 above)
Comment (pushfoo) [reply doc/example_code/how_to_examples/index.rst:408]: My intent with heading names was to reflect heading hierarchy from left to right, like a file path. To my knowledge, sphinx doesn't support this sort of nesting outside of toctree declarations, but I might be wrong.

I'd like to switch the other instances to follow this pattern, but I'm not sure it's appropriate to do so in this PR.

Your thoughts?

File: doc/programming_guide/sprites/spritelists.rst (hunk 4 above)
Comment (pushfoo) [reply doc/programming_guide/sprites/spritelists.rst]: I've removed the exclamation point.

File: doc/example_code/how_to_examples/index.rst (hunk 1 above)
Comment (pvcraven) [reply doc/example_code/how_to_examples/index.rst:408]: I think it is inconsistent to have examples_sprites and camera_examples. I get the header-thing but this really sticks out to me for some reason.

Comment (pushfoo) [reply doc/example_code/how_to_examples/index.rst:408]: I think you're misunderstanding: I am proposing we rename everything to `examples_{subtype}`

Comment (pushfoo) [reply doc/example_code/how_to_examples/index.rst:408]: I'll put this in a separate PR per discord discussion.

File: arcade/examples/sprite_minimal.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        self.sprites = None
+        self.setup()
+
+    def setup(self):
+        # 1. Create the SpriteList
+        self.sprites = arcade.SpriteList()
+
+        # 2. Create & append your Sprite instance to the SpriteList
+        self.circle = arcade.SpriteCircle(30, arcade.color.WHITE)  # 30 pixel radius circle
+        self.circle.position = self.width // 2, self.height // 2  # Put it in the middle
```
Comment (gran4) [inline arcade/examples/sprite_minimal.py:25]: Does it really need to be self.circle. I think if I was a beginner I would do that every time I made a sprite. It seems to be convention on arcade stuff, is there a reason for that?

**Review by pvcraven — APPROVED** [review]

**Pull request merged**